### ISSUE TITLE
question about ti cc1352p-2 

### ISSUE THREAD

**Issue by flower87lucky** [Koenkk/zigbee2mqtt#2162]
)
zigbee2mqtt:info  2019-10-19T12:50:02: Starting zigbee-herdsman...
zigbee2mqtt:error 2019-10-19T12:50:09: Error while starting zigbee-herdsman
zigbee2mqtt:error 2019-10-19T12:50:09: Failed to start zigbee
zigbee2mqtt:error 2019-10-19T12:50:09: Exiting...
zigbee2mqtt:error 2019-10-19T12:50:09: Error: SRSP - SYS - version after 6000ms
    at Timeout.object.timer.setTimeout [as _onTimeout] (/app/node_modules/zigbee-herdsman/dist/utils/waitress.js:44:24)
    at ontimeout (timers.js:436:11)
    at tryOnTimeout (timers.js:300:5)
    at listOnTimeout (timers.js:263:5)
    at Timer.processTimers (timers.js:223:10)
npm
 ERR! code
 ELIFECYCLE
npm 
ERR! errno 1
npm
 ERR! zigbee2mqtt@1.6.0 start: `node index.js`
npm 
ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the zigbee2mqtt@1.6.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm
 ERR!     /root/.npm/_logs/2019-10-19T12_50_09_649Z-debug.log
2019-10-19T20:50:09: PM2 log: App [npm:0] exited with code [1] via signal [SIGINT]
2019-10-19T20:50:09: PM2 log: App [npm:0] starting in -fork mode-
2019-10-19T20:50:09: PM2 log: App

**Comment by flower87lucky** [comment]
serial:
/dev/ttyACM1
/dev/ttyACM0
/dev/serial/by-id/usb-Texas_Instruments_XDS110__02.03.00.18__Embed_with_CMSIS-DAP_L43000GM-if03
/dev/ttyAMA0
/dev/serial/by-id/usb-Texas_Instruments_XDS110__02.03.00.18__Embed_with_CMSIS-DAP_L43000GM-if00
input:

**Comment by flower87lucky** [comment]
"serial": {
    "port": "/dev/ttyACM0"
  },
  "advanced": {
    "pan_id": 6758,
    "channel": 15,

**Comment by hdo** [comment]
I got the same error message with my 2652R board :-(
I also tried to change panid and channel in my configuration.

**Comment by bertran1** [comment]
I have also this problem if I use a static channel in the config. I have don't use a static config, I don't have this problem.

**Comment by Koenkk** [comment]
Can you try with 

```js
"serial": {
"port": null,
},
```

**Comment by flower87lucky** [comment]
> Can you try with
> 
> ```js
> "serial": {
> "port": null,
> },
> ```

Thanks.null is not allowed in Zigbee2mqtt Hass.io Add-on .I used Zigbee2mqtt Hass.io Add-on,maybe it may not support at present. I try to install ZigBee 2mqtt directly

**Comment by flower87lucky** [comment]
> Can you try with
> 
> ```js
> "serial": {
> "port": null,
> },
> ```

I'm worried if I flash the firmware in the wrong way?
After cc1352p-2 flashed the firmware, I press reset key and it has no effect.

**Comment by Koenkk** [comment]
- Can you try the steps from https://www.zigbee2mqtt.io/information/zigbee2mqtt_fails_to_start.html
- Can you make a screenshot when flashing via Flash programmer v2?

**Comment by hdo** [comment]
> * Can you try the steps from https://www.zigbee2mqtt.io/information/zigbee2mqtt_fails_to_start.html
> * Can you make a screenshot when flashing via Flash programmer v2?

@Koenkk 

I still have the problem that i need to soft reset first (via button press) to start the firmware.

Do you know how to disable this behaviour?

Little change request:

Let the green LED light up when firmware has loaded for quick check ;-)

**Comment by flower87lucky** [comment]
> * Can you try the steps from https://www.zigbee2mqtt.io/information/zigbee2mqtt_fails_to_start.html
> * Can you make a screenshot when flashing via Flash programmer v2?

![Image 2](https://user-images.githubusercontent.com/45326473/67181126-41131580-f40e-11e9-8988-2c0cee6603d3.png)

```
pi@raspberrypi:~ $ ls -l /dev/serial/by-id
total 0
lrwxrwxrwx 1 root root 13 Oct 21 14:23 usb-Texas_Instruments_XDS110__02.03.00.18__Embed_with_CMSIS-DAP_L43000GM-if00 -> ../../ttyACM0
lrwxrwxrwx 1 root root 13 Oct 21 14:23 usb-Texas_Instruments_XDS110__02.03.00.18__Embed_with_CMSIS-DAP_L43000GM-if03 -> ../../ttyACM1
```
```
pi@raspberrypi:~ $ test -w /dev/ttyACM0 && echo success || echo failure
success
pi@raspberrypi:~ $ test -w /dev/ttyACM1 && echo success || echo failure
success
pi@raspberrypi:~ $ 
```

**Comment by flower87lucky** [comment]
> * Can you try the steps from https://www.zigbee2mqtt.io/information/zigbee2mqtt_fails_to_start.html
> * Can you make a screenshot when flashing via Flash programmer v2?

![DE2C8A3E876723407674E0E21AA6EAD3](https://user-images.githubusercontent.com/45326473/67181728-0ad69580-f410-11e9-9b45-c52908123b23.JPG)
When I connect to raspiberry pi 4b+ ,only the green led on.

**Comment by Koenkk** [comment]
@fredrikgk chip revision C is an older revision, right? Is it binary compatible with the E revision?

**Comment by flower87lucky** [comment]
> @fredrikgk chip revision C is an older revision, right? Is it binary compatible with the E revision?

@Koenkk   How to fix binary to compatible，thanks.

**Comment by fredrikgk** [comment]
@flower87lucky , @Koenkk , IC revision C is deprecated and not supported. SW is not binary compatible between rev. C and rev. E. 

The only alternative is to order a new LaunchPad.

**Comment by hdo** [comment]
@fredrikgk 

BTW Do you have an idea why i need to soft reset (via button press) each time after USB plug to start the firmware?

**Comment by flower87lucky** [comment]
> @fredrikgk
> 
> BTW Do you have an idea why i need to soft reset (via button press) each time after USB plug to start the firmware?

I agree you.

**Comment by Koenkk** [comment]
@fredrikgk thanks, I will close this.

**Comment by james-fry** [comment]
> @fredrikgk
> 
> BTW Do you have an idea why i need to soft reset (via button press) each time after USB plug to start the firmware?

I got caught out with the need for soft reset too. 
(and thanks for your assistance with this in the z2m issue @hdo)

I agree with your suggestion that either/both of the following would be good:
- No need for soft reset on plug in of device
- One of the LEDs on the MCU side will be used to indicate status or "success"

@Koenkk are either of these doable? should I raise an issue on the zstack fw repo?

**Comment by hdo** [comment]
@james-fry 

This is not a simple task.
I compiled a custom firmware version with LED support and that is also not a good indicator for this problem. I learned the firmware is loading correctly. However the culprit is the onbard debugger of the launchpad. On my Intel NUC i need the soft reset but not on my raspberry pi. There seems to be an issue with the usb handshake or something like this. I don't have any issues using the UART so that's why i think it's the usb part. The UART have other issues though (sending leading ZERO in some circumstances).

**Comment by james-fry** [comment]
Thanks for the info @hdo 
I am also on intel so maybe that is why I also need the reset, but its not a big deal when you know you need to do this.

**Comment by fredrikgk** [comment]
@hdo Do you have to press the reset button also when the USB cable has been disconnected for a long time?

**Comment by fredrikgk** [comment]
> I compiled a custom firmware version with LED support and that is also not a good indicator for this problem. I learned the firmware is loading correctly. However the culprit is the onbard debugger of the launchpad. On my Intel NUC i need the soft reset but not on my raspberry pi. There seems to be an issue with the usb handshake or something like this. I don't have any issues using the UART so that's why i think it's the usb part. The UART have other issues though (sending leading ZERO in some circumstances).

This is strange, as the reset button only resets the CC1352P, not the debugger. 

How is your LaunchPad configured (jumpers) when you are "using UART"?

**Comment by hdo** [comment]
> @hdo Do you have to press the reset button also when the USB cable has been disconnected for a long time?

I have to soft reset each time i plug in the power cable.

>This is strange, as the reset button only resets the CC1352P, not the debugger.

I not sure about that.

> How is your LaunchPad configured (jumpers) when you are "using UART"?

I removed the TX/RX jumpers.

It could also work without removing those jumpers but i noticed some preceeding '0'-byte which disturbs the communication, at least in my environment.

Also see: [0-byte](http://e2e.ti.com/support/wireless-connectivity/sub-1-ghz/f/156/t/842593?CCS-SIMPLELINK-CC13X2-26X2-SDK-Coprocessor-TI-15-4-adds-byte-0-as-first-byte-before-SYS-RESET-IND-response-for-SYS-RESET-REQ-when-communicating-directly-with-the-cc13x2-DIO12-13-)

**Comment by a-bailey** [comment]
I seem to have a B revision of the launchpad board is this supposed to work? 

pi@raspberrypi:~ $ test -w /dev/ttyACM0 && echo success || echo failure
failure
pi@raspberrypi:~ $ test -w /dev/ttyACM1 && echo success || echo failure
failure

getting failure on both write tests.

**Comment by dariuszqrek** [comment]
I have board revision B, chip revision E and have same error - zigbee2mqtt won't start.
what could be a problem ?

**Comment by timdonovanuk** [comment]
How do you know what revision board you've got? Going to be a bit annoyed if I wasted $50 on a board that isn't supported when the z2m docs don't mention that only certain revs work :(

**Comment by dariuszqrek** [comment]
It looks like i was wrong, everything is working well now.
There was need to change pan_id in configuration.
As far as i know TI is sending only correct revision now (Board revision B
with chip revision E)


niedz., 22 gru 2019, 18:39 użytkownik timdonovanuk <notifications@github.com>
napisał:

**Comment by CodeFinder2** [comment]
I am **also having the "need to push the board's reset button before I can start Z2M" issue** (CC1352P-2) with my Odroid XU4. Connected via USB, jumpers are on their factory defaults.

Am I right that there's nothing we can do about it now? Does anyone know whether it can be fixed at all?

For posterity, this is my working config:
```yaml
homeassistant: false
permit_join: true
mqtt:
  base_topic: zigbee2mqtt
  server: 'mqtt://localhost'
  user: YOUR_USERNAME
  password: YOUR_PASSWORD
serial:
  port: null
advanced:
  pan_id: 6755
  baudrate: 115200
  report: true
  rtscts: false
  channel: 20
  log_level: debug
  network_key:
    - 1
    - 3
    - 5
    - 7
    - 9
    - 11
    - 13
    - 15
    - 0
    - 2
    - 4
    - 6
    - 8
    - 10
    - 12
    - 13
```

Hardware (board revision B, printed on both sides of my board):
```shell
$ ls -l /dev/serial/by-id
lrwxrwxrwx 1 root root 13 Dez 24 01:39 usb-Texas_Instruments_XDS110__03.00.00.05__Embed_with_CMSIS-DAP_L43001PF-if00 -> ../../ttyACM0
lrwxrwxrwx 1 root root 13 Dez 24 01:39 usb-Texas_Instruments_XDS110__03.00.00.05__Embed_with_CMSIS-DAP_L43001PF-if03 -> ../../ttyACM1

$ test -w /dev/ttyACM0 && echo success || echo failure
success
$ test -w /dev/ttyACM1 && echo success || echo failure
success

$ npm start

> zigbee2mqtt@1.8.0-dev start /opt/zigbee2mqtt
> node index.js

zigbee2mqtt:info  2019-12-24 01:50:50: Logging to console and directory: '/opt/zigbee2mqtt/data/log/2019-12-24.01-50-50'
zigbee2mqtt:debug 2019-12-24 01:50:50: Removing old log directory '/opt/zigbee2mqtt/data/log/2019-12-24.01-11-43'
zigbee2mqtt:debug 2019-12-24 01:50:50: Loaded state from file /opt/zigbee2mqtt/data/state.json
zigbee2mqtt:info  2019-12-24 01:50:50: Starting zigbee2mqtt version 1.8.0-dev (commit #66b3bcb)
zigbee2mqtt:info  2019-12-24 01:50:50: Starting zigbee-herdsman...
zigbee2mqtt:debug 2019-12-24 01:50:50: Using zigbee-herdsman with settings: '{"network":{"panID":6755,"extendedPanID":[221,221,221,221,221,221,221,221],"channelList":[20],"networkKey":"HIDDEN"},"databasePath":"/opt/zigbee2mqtt/data/database.db","databaseBackupPath":"/opt/zigbee2mqtt/data/database.db.backup","backupPath":"/opt/zigbee2mqtt/data/coordinator_backup.json","serialPort":{"baudRate":115200,"rtscts":false,"path":null}}'
zigbee2mqtt:info  2019-12-24 01:50:51: zigbee-herdsman started
zigbee2mqtt:info  2019-12-24 01:50:51: Coordinator firmware version: '{"type":"zStack3x0","meta":{"transportrev":2,"product":1,"majorrel":2,"minorrel":7,"maintrel":1,"revision":20191106}}'
zigbee2mqtt:debug 2019-12-24 01:50:51: Zigbee network parameters: {"panID":6755,"extendedPanID":"0xdddddddddddddddd","channel":20}
zigbee2mqtt:info  2019-12-24 01:50:51: Currently 1 devices are joined:
zigbee2mqtt:info  2019-12-24 01:50:51: 0x086bd7fffee976a6 (0x086bd7fffee976a6): 2AJZ4KPKEY - Konke Multi-function button (EndDevice)
zigbee2mqtt:warn  2019-12-24 01:50:51: `permit_join` set to  `true` in configuration.yaml.
zigbee2mqtt:warn  2019-12-24 01:50:51: Allowing new devices to join.
zigbee2mqtt:warn  2019-12-24 01:50:51: Set `permit_join` to `false` once you joined all devices.
zigbee2mqtt:info  2019-12-24 01:50:51: Zigbee: allowing new devices to join.
zigbee2mqtt:info  2019-12-24 01:50:51: Connecting to MQTT server at mqtt://localhost
zigbee2mqtt:info  2019-12-24 01:50:51: Connected to MQTT server
zigbee2mqtt:info  2019-12-24 01:50:51: MQTT publish: topic 'zigbee2mqtt/bridge/state', payload 'online'
zigbee2mqtt:info  2019-12-24 01:50:51: MQTT publish: topic 'zigbee2mqtt/0x086bd7fffee976a6', payload '{"linkquality":99}'
zigbee2mqtt:info  2019-12-24 01:50:51: MQTT publish: topic 'zigbee2mqtt/bridge/config', payload '{"version":"1.8.0-dev","commit":"66b3bcb","coordinator":{"type":"zStack3x0","meta":{"transportrev":2,"product":1,"majorrel":2,"minorrel":7,"maintrel":1,"revision":20191106}},"log_level":"debug","permit_join":true}'
```

Note: the reset button is the button next to the micro USB port and `port: null` correctly auto-detected my board.

**Comment by TASSDevon** [comment]
> How do you know what revision board you've got? Going to be a bit annoyed if I wasted $50 on a board that isn't supported when the z2m docs don't mention that only certain revs work :(

The board revision is printed on the back of the board "HW Rev", the chip version can be read out using the Uniflash tool.

For reference, I bought mine on Mouser (Dutch version of the site, but the board came from Texas). I have a HW Rev B and Chip Rev E (2.1).

Still need to test it. I read that you need to re-pair all devices because of the bump to Zigbee stack version 3, is this correct? Perhaps this should be stated on the supported adapters page.

Currently I have a CC2531 hooked up to a Beaglebone Black running native zigbee2mqtt, other than changing the serial:port to whatever the TI board enumerates as, anything else I should know or take into consideration before doing the big ol' switcharoo (re-pair-a-roo?)

**Comment by CodeFinder2** [comment]
@TASSDevon no it does not auto-detect an external antenna. You have to resolder a tiny capacitor on the board to use the external antenna...unfortunately.

See: https://e2e.ti.com/support/wireless-connectivity/other-wireless/f/667/t/791444?LAUNCHXL-CC1352P-Clarification-of-ports-for-external-antennas

Nothing special that I am aware of, you should consider when moving to this chip, except the re-pairing.

**Comment by timdonovanuk** [comment]
See https://github.com/Koenkk/zigbee-herdsman/issues/113#issuecomment-569619621 - there is hope that repairing all devices is not necessary. I'm holding off moving to the CC1352P-2 until confirmed. I have 60+ devices and repairing will take an entire weekend :)

**Comment by CodeFinder2** [comment]
@Koenkk may (some of) [these information](https://github.com/Koenkk/zigbee2mqtt/issues/2162#issuecomment-570286663) be relevant for the docs (=> update)?

**Comment by james-fry** [comment]
> See [Koenkk/zigbee-herdsman#113 (comment)](https://github.com/Koenkk/zigbee-herdsman/issues/113#issuecomment-569619621) - there is hope that repairing all devices is not necessary. I'm holding off moving to the CC1352P-2 until confirmed. I have 60+ devices and repairing will take an entire weekend :)

Unless something changed, moving from a z-stack 1.2.x to z-stack 3.x coordinator requires a re-pair of all devices.

**Comment by CodeFinder2** [comment]
> > See [Koenkk/zigbee-herdsman#113 (comment)](https://github.com/Koenkk/zigbee-herdsman/issues/113#issuecomment-569619621) - there is hope that repairing all devices is not necessary. I'm holding off moving to the CC1352P-2 until confirmed. I have 60+ devices and repairing will take an entire weekend :)
> 
> Unless something changed, moving from a z-stack 1.2.x to z-stack 3.x coordinator requires a re-pair of all devices.

Have you tried this yourself (and failed)? As it is still said in the linked issue by Koenkk ("I think this can already be done without any code changes [...]."), it may be possible ...

**Comment by TASSDevon** [comment]
Indeed, has anyone actually already tried this so they can confirm it works or not (and perhaps for which devices it does not work).

I will give it a try on Saturday and report back.

**Comment by jrhansen** [comment]
What is the solution to the problem? I have the exact same error with a brand new CC1352P-2 (board rev. B, chip rev. E):

```
pi@rpi4:~/zigbee2mqtt $ npm start

> zigbee2mqtt@1.8.0 start /home/pi/zigbee2mqtt
> node index.js

zigbee2mqtt:info  2020-01-14 18:07:33: Logging to console and directory: '/home/pi/zigbee2mqtt/data/log/2020-01-14.18-07-33'
zigbee2mqtt:info  2020-01-14 18:07:34: Starting zigbee2mqtt version 1.8.0 (commit #da4d26a)
zigbee2mqtt:info  2020-01-14 18:07:34: Starting zigbee-herdsman...
zigbee2mqtt:error 2020-01-14 18:07:41: Error while starting zigbee-herdsman
zigbee2mqtt:error 2020-01-14 18:07:41: Failed to start zigbee
zigbee2mqtt:error 2020-01-14 18:07:41: Exiting...
zigbee2mqtt:error 2020-01-14 18:07:41: Error: SRSP - SYS - version after 6000ms
    at Timeout._onTimeout (/home/pi/zigbee2mqtt/node_modules/zigbee-herdsman/dist/utils/waitress.js:44:24)
    at listOnTimeout (internal/timers.js:531:17)
    at processTimers (internal/timers.js:475:7)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! zigbee2mqtt@1.8.0 start: `node index.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the zigbee2mqtt@1.8.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/pi/.npm/_logs/2020-01-14T17_07_41_611Z-debug.log
```

The ports are writeable, and I changed the pan ID. The setup works perfectly, if I re-connect the CC2531.

The mentioned log file contains:

```
0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/node', '/usr/bin/npm', 'start' ]
2 info using npm@6.13.4
3 info using node@v12.14.1
4 verbose run-script [ 'prestart', 'start', 'poststart' ]
5 info lifecycle zigbee2mqtt@1.8.0~prestart: zigbee2mqtt@1.8.0
6 info lifecycle zigbee2mqtt@1.8.0~start: zigbee2mqtt@1.8.0
7 verbose lifecycle zigbee2mqtt@1.8.0~start: unsafe-perm in lifecycle true
8 verbose lifecycle zigbee2mqtt@1.8.0~start: PATH: /usr/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/home/pi/zigbee2mqtt/node_modules/.bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/local/games:/usr/games
9 verbose lifecycle zigbee2mqtt@1.8.0~start: CWD: /home/pi/zigbee2mqtt
10 silly lifecycle zigbee2mqtt@1.8.0~start: Args: [ '-c', 'node index.js' ]
11 silly lifecycle zigbee2mqtt@1.8.0~start: Returned: code: 1  signal: null
12 info lifecycle zigbee2mqtt@1.8.0~start: Failed to exec start script
13 verbose stack Error: zigbee2mqtt@1.8.0 start: `node index.js`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:332:16)
13 verbose stack     at EventEmitter.emit (events.js:223:5)
13 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:223:5)
13 verbose stack     at maybeClose (internal/child_process.js:1021:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)
14 verbose pkgid zigbee2mqtt@1.8.0
15 verbose cwd /home/pi/zigbee2mqtt
16 verbose Linux 4.19.75-v7l+
17 verbose argv "/usr/bin/node" "/usr/bin/npm" "start"
18 verbose node v12.14.1
19 verbose npm  v6.13.4
20 error code ELIFECYCLE
21 error errno 1
22 error zigbee2mqtt@1.8.0 start: `node index.js`
22 error Exit status 1
23 error Failed at the zigbee2mqtt@1.8.0 start script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]

```
Any help is really appreciated! :-)

**Comment by CodeFinder2** [comment]
@jrhansen  connect the board and then press the reset button...

**Comment by jrhansen** [comment]
@CodeFinder2 Already tried it, but tried it again with the same error. The correct sequence is to power the board, and THEN momentarily press the button close to the USB connector, right? Not press the button while connecting the board? Should I see any activity on the LEDs when resetting the board?

**Comment by TASSDevon** [comment]
I just did the switch, used port 'null' for autodetect, with the database restore 'trick' and changing the pad_id as stated on the config page, here's what I found:

- Eurotronics Zigbee valve automatically got picked up, no re-pairing required
- Xiaomi Mija sensors all needed to be re-paired but it was SO much easier than with the old CC2351 (with that one I usually had to try it a couple of times close to the coordinator for them to pair, here it went from first try even across my house)
- 2 of my 5 ikea tradfri outlets re-paired immediately, for the other 3 I had to explicitly remove them from the network first for some reason
- CC2351 router repaired immediately

The signal quality before:
<img width="1176" alt="Screenshot 2020-01-02 at 18 36 23" src="https://user-images.githubusercontent.com/6917072/72376134-c0fb0400-370d-11ea-8ad5-3a6036b48edf.png">

The signal quality after:
<img width="1173" alt="Screenshot 2020-01-14 at 20 37 07" src="https://user-images.githubusercontent.com/6917072/72376159-c9533f00-370d-11ea-8288-87ba0d988e76.png">

So a nice little bit of improvement there, I should get an SMA antenna to increase that even more, anyone can recommend me one?

If anyone else has questions about switching to the TI hardware, let me know. Aside from the 3 tradri outlets and figuring out how to solve the situation that they didn't want to repair, it took me perhaps an hour or so to do everything.

**Comment by jrhansen** [comment]
“database restore 'trick'”? :-)

**Comment by CodeFinder2** [comment]
> “database restore 'trick'”? :-)

https://github.com/Koenkk/zigbee-herdsman/issues/113#issuecomment-569619621

**Comment by CodeFinder2** [comment]
> [...] 
> So a nice little bit of improvement there, I should get an SMA antenna to increase that even more, anyone can recommend me one? [...] 

I bought this one (because it was also labeled "Zigbee") and it works good, see my post above: https://www.digikey.de/product-detail/de/linx-technologies-inc/ANT-2.4-CW-HW-SMA/ANT-2.4-CW-HW-SMA-ND/2391322 (8€)

Don't know if any similar (Cheap Chinese 😂) SMA 2.4 GHz antenna would do the same job. 😁

**Comment by remkolems** [comment]
> What is the solution to the problem? I have the exact same error with a brand new CC1352P-2 (board rev. B, chip rev. E):
> 
> ```
> pi@rpi4:~/zigbee2mqtt $ npm start
> 
> > zigbee2mqtt@1.8.0 start /home/pi/zigbee2mqtt
> > node index.js
> 
> zigbee2mqtt:info  2020-01-14 18:07:33: Logging to console and directory: '/home/pi/zigbee2mqtt/data/log/2020-01-14.18-07-33'
> zigbee2mqtt:info  2020-01-14 18:07:34: Starting zigbee2mqtt version 1.8.0 (commit #da4d26a)
> zigbee2mqtt:info  2020-01-14 18:07:34: Starting zigbee-herdsman...
> zigbee2mqtt:error 2020-01-14 18:07:41: Error while starting zigbee-herdsman
> zigbee2mqtt:error 2020-01-14 18:07:41: Failed to start zigbee
> zigbee2mqtt:error 2020-01-14 18:07:41: Exiting...
> zigbee2mqtt:error 2020-01-14 18:07:41: Error: SRSP - SYS - version after 6000ms
>     at Timeout._onTimeout (/home/pi/zigbee2mqtt/node_modules/zigbee-herdsman/dist/utils/waitress.js:44:24)
>     at listOnTimeout (internal/timers.js:531:17)
>     at processTimers (internal/timers.js:475:7)
> npm ERR! code ELIFECYCLE
> npm ERR! errno 1
> npm ERR! zigbee2mqtt@1.8.0 start: `node index.js`
> npm ERR! Exit status 1
> npm ERR! 
> npm ERR! Failed at the zigbee2mqtt@1.8.0 start script.
> npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
> 
> npm ERR! A complete log of this run can be found in:
> npm ERR!     /home/pi/.npm/_logs/2020-01-14T17_07_41_611Z-debug.log
> ```
> 
> The ports are writeable, and I changed the pan ID. The setup works perfectly, if I re-connect the CC2531.
> 
> The mentioned log file contains:
> 
> ```
> 0 info it worked if it ends with ok
> 1 verbose cli [ '/usr/bin/node', '/usr/bin/npm', 'start' ]
> 2 info using npm@6.13.4
> 3 info using node@v12.14.1
> 4 verbose run-script [ 'prestart', 'start', 'poststart' ]
> 5 info lifecycle zigbee2mqtt@1.8.0~prestart: zigbee2mqtt@1.8.0
> 6 info lifecycle zigbee2mqtt@1.8.0~start: zigbee2mqtt@1.8.0
> 7 verbose lifecycle zigbee2mqtt@1.8.0~start: unsafe-perm in lifecycle true
> 8 verbose lifecycle zigbee2mqtt@1.8.0~start: PATH: /usr/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/home/pi/zigbee2mqtt/node_modules/.bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/local/games:/usr/games
> 9 verbose lifecycle zigbee2mqtt@1.8.0~start: CWD: /home/pi/zigbee2mqtt
> 10 silly lifecycle zigbee2mqtt@1.8.0~start: Args: [ '-c', 'node index.js' ]
> 11 silly lifecycle zigbee2mqtt@1.8.0~start: Returned: code: 1  signal: null
> 12 info lifecycle zigbee2mqtt@1.8.0~start: Failed to exec start script
> 13 verbose stack Error: zigbee2mqtt@1.8.0 start: `node index.js`
> 13 verbose stack Exit status 1
> 13 verbose stack     at EventEmitter.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:332:16)
> 13 verbose stack     at EventEmitter.emit (events.js:223:5)
> 13 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
> 13 verbose stack     at ChildProcess.emit (events.js:223:5)
> 13 verbose stack     at maybeClose (internal/child_process.js:1021:16)
> 13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)
> 14 verbose pkgid zigbee2mqtt@1.8.0
> 15 verbose cwd /home/pi/zigbee2mqtt
> 16 verbose Linux 4.19.75-v7l+
> 17 verbose argv "/usr/bin/node" "/usr/bin/npm" "start"
> 18 verbose node v12.14.1
> 19 verbose npm  v6.13.4
> 20 error code ELIFECYCLE
> 21 error errno 1
> 22 error zigbee2mqtt@1.8.0 start: `node index.js`
> 22 error Exit status 1
> 23 error Failed at the zigbee2mqtt@1.8.0 start script.
> 23 error This is probably not a problem with npm. There is likely additional logging output above.
> 24 verbose exit [ 1, true ]
> ```
> 
> Any help is really appreciated! :-)

I also have the same issues... 

Started out with 2x CC2531 w/ SMA antenna (coordinator and router), but signal strength was far from ideal. Therefore, "upgraded" to a brand new CC1352P-2 (in future with the SMA antenna). Upgrade part seems to be very tricky to say the least. Platform: Home-Assistant w/ add-on ZigBee2MQTT (dev branch also tested) and Mosquitto broker (official). No luck until now. IMHO: hold off the upgrade part until it is/has been resolved. BTW the null trick is not permitted with the Home-Assistant ZigBee2MQTT addon. 

Will update this post if solved.

**Comment by CodeFinder2** [comment]
@remkolems  I got it working already with HA and hassio addon (just as a test / proof of concept before I really use the CC1352 as my daily driver).

This should give you a working start (Hassio addon zigbee2mqtt):
```json
{
  "data_path": "/share/zigbee2mqtt",
  "devices": "devices.yaml",
  "groups": "groups.yaml",
  "homeassistant": true,
  "permit_join": false,
  "mqtt": {
    "base_topic": "zigbee2mqtt",
    "server": "mqtt://core-mosquitto",
    "user": "YOUR_USERNAME",
    "password": "YOUR_PASSWORD"
  },
  "serial": {
    "port": "/dev/ttyACM0"
  },
  "advanced": {
    "baudrate": 115200,
    "log_level": "info",
    "rtscts": false,
    "report": true,
    "pan_id": 6755,
    "last_seen": "ISO_8601",
    "channel": 25,
    "availability_blacklist": [],
    "network_key": [ YOUR_KEY_HERE ]
  },
  "ban": [],
  "whitelist": [],
  "queue": {},
  "socat": {
    "enabled": false,
    "master": "pty,raw,echo=0,link=/dev/ttyZ2M,mode=777",
    "slave": "tcp-listen:8485,keepalive,nodelay,reuseaddr,keepidle=1,keepintvl=1,keepcnt=5",
    "restartdelay": 1,
    "initialdelay": 1,
    "options": "-d -d",
    "log": false
  }
}
```

PS: the CC2531s are really cheap/simple boards, see my post [here](https://github.com/Koenkk/zigbee2mqtt/issues/2162#issuecomment-570286663) regarding signal strength.

**Comment by CodeFinder2** [comment]
> @CodeFinder2 Already tried it, but tried it again with the same error. The correct sequence is to power the board, and THEN momentarily press the button close to the USB connector, right? Not press the button while connecting the board? Should I see any activity on the LEDs when resetting the board?

Exactly:
- Connect it first.
- After that (you just need to this once, the delay after you connected your board doesn't matter), press the button next ot the USB connector once (momentarily/shortly).
- On my board, there's neither LED activitiy nor a hearable "click" when pushing the button shortly.
- Then start zigbee2mqtt.

This working for me (tested with dev branch but should also work with v1.8.0).

**Comment by remkolems** [comment]
@CodeFinder2 Thank you! It is working now! Looks like we almost had the same config! I copied some missing items from your advanced section and the reset button!

For reference purposes: my previous ZigBee2MQTT HA addon config.
```
{
  "data_path": "/share/zigbee2mqtt",
  "devices": "devices.yaml",
  "groups": "groups.yaml",
  "homeassistant": true,
  "permit_join": false,
  "mqtt": {
    "base_topic": "zigbee2mqtt",
    "server": "mqtt://YOUR_BROKER:1883",
    "user": "YOUR_USERNAME",
    "password": "YOUR_PASSWORD
  },
  "serial": {
    "port": "/dev/ttyACM1"
  },
  "advanced": {
    "pan_id": 6755,
    "channel": 25,
    "network_key": [
      1,
      3,
      5,
      7,
      9,
      11,
      13,
      15,
      0,
      2,
      4,
      6,
      8,
      10,
      12,
      13
    ],
    "availability_blacklist": []
  },
  "ban": [],
  "whitelist": [],
  "queue": {},
  "socat": {
    "enabled": false,
    "master": "pty,raw,echo=0,link=/dev/ttyZ2M,mode=777",
    "slave": "tcp-listen:8485,keepalive,nodelay,reuseaddr,keepidle=1,keepintvl=1,keepcnt=5",
    "restartdelay": 1,
    "initialdelay": 1,
    "options": "-d -d",
    "log": false
  }
}
```

**Comment by docBliny** [comment]
I'm in the middle of a broken setup, too. I was getting the `Error: SRSP - SYS - version after 6000ms` error until I made the following changes (not sure if all were needed):
* I'm running in Proxmox, so I changed from USB Vendor Device ID to USB port assignment
* Rebooted the whole Proxmox computer
* Hit the reset button on board

Since "null" isn't supported via HA, I'm running with `"port": "/dev/ttyACM1"` and I got a connection in the end. However, I still haven't managed to get any traffic to show up. I restored the `database.db` and it looks like they're all getting loaded, but nothing past that yet.

**Comment by remkolems** [comment]
> I'm in the middle of a broken setup, too. I was getting the `Error: SRSP - SYS - version after 6000ms` error until I made the following changes (not sure if all were needed):
> 
> * I'm running in Proxmox, so I changed from USB Vendor Device ID to USB port assignment
> * Rebooted the whole Proxmox computer
> * Hit the reset button on board
> 
> Since "null" isn't supported via HA, I'm running with `"port": "/dev/ttyACM1"` and I got a connection in the end. However, I still haven't managed to get any traffic to show up. I restored the `database.db` and it looks like they're all getting loaded, but nothing past that yet.
> 
> Edit: Well, I tried reconnecting one of my Xiaomi buttons and it started working after I did the reconnect. This won't be fun if I need to reconnect them all (~70 devices).

That is what I had to do also (still doing to be honest). A real pain in the a s s ....

**Comment by docBliny** [comment]
Yup, just spent the evening resetting everything. Found several water sensors that had run out of battery while I was at it, so something positive. Also looked like the Aqara water sensors would connect just by activating them twice (short press). The first connect would fail with `Failed to interview 'water_leak_15', device has not successfully been paired` but the second attempt would register it (without the long hold to "reset").

**Comment by jrhansen** [comment]
1. Powered up an old RPi3 with a fresh Raspbian lite SD card.
2. Installed Mosquitto.
3. Followed the Z2M installation instructions.
4. Setup the correct port (and initially forgot to set a different pan_id than my existing Zigbee network).
5. On first 'npm start' I got the 'SRSP - SYS - version after 6000ms' error.
6. Pushed the reset button.
7. 'npm start' again -> Same error.
8. Disconnected the CC1352P-2 from the USB port.
9. Reconnected the board.
10. 'npm start' again -> Same error.
11. Pushed the reset button.
12. 'npm start' -> Error saying that the pan_is already in use!
13. Changed the pan_id.
14. 'npm start' -> Z2M now starts up correctly :-)

So, it seems that things have started to work now even though I tried the reset button and all kinds of other stuff yesterday to no avail. I tried to break it again by reconnecting the board (in different USB ports), rebooting, pressing the reset button, changing the pan_id, going from '/dev/ACM0' to 'null' in the port setup, but it seems unbreakable now. I can join new devices without trouble, and I don't have to press the reset button again - even after a power cycle (??).

Now, I will go back to implementing the board in my existing setup with the risk of having the rejoin everything.

When I did experiments yesterday with my existing network, when the CC1352P-2 failed I had to go back to the CC2531, and I had to rejoin all devices even though I had backed up the 'data' folder. So...

Should I backup other folders than the 'data' folder? The entire Z2M folder, or something else in the system?

**Comment by CodeFinder2** [comment]
@jrhansen  IMHO, the `data` directory is sufficient. But going forward and backward between z stack versions (1.2 vs 3) can be tricky / may require re-pairing (regardless of a backup).

My CC1352P-2 is now running as my daily driver for a few days now with Z2M 1.9.0 from HassIO and everything working fine so far. No issues and (re-) pairing of all my devices was SO smooth, it even made kinda fun. 😂

**Comment by TheStigh** [comment]
> 
> 
> @jrhansen IMHO, the `data` directory is sufficient. But going forward and backward between z stack versions (1.2 vs 3) can be tricky / may require re-pairing (regardless of a backup).
> 
> My CC1352P-2 is now running as my daily driver for a few days now with Z2M 1.9.0 from HassIO and everything's working fine so far. No issues and (re-) pairing of all my devices was SO smooth (I did not even try w/o re-pairing as I wanted a fresh install), it even made kinda fun. 😂

@CodeFinder2 
I've just started playing with z2m and installed a standalone docker installation on a RPi4 with the CC2531. Currently I'm waiting for my CC1352P-2, and when it arrives I would like to move my z2m to my HA hardware running on a NUC i7.

My question to you would be; should I run the z2m HA addon or would I benefit from running z2m as a docker install on the same hardware?

**Comment by CodeFinder2** [comment]
> [...]
> My question to you would be; should I run the z2m HA addon or would I benefit from running z2m as a docker install on the same hardware?

The advantage of the hassio addon is it's simplicity in installing / managing it (single web instance: HA). Behind the scenes, it's a docker container as well (you might want to take a look at the "Portrainer" addon).

I am using the hassio addon and I'm quite happy with it. Also having features like 'secrets' and accessing the log in HA should be taken into consideration.

The disadvantage is mainly that you are depending on the release cycle of the addon (although this is normally the same as for Z2M). Also switching between stable/dev might be a bit more complicated sometimes.

So, as usual: it depends. 🙈 I would suggest that you start using the hassio addon first. By the way, you can also use somewhat both: install both, do all the pairing and configuration you require in the addon. Then you should be fine in copying the config over to the docker instance. Of cause you shouldn't run both simultaneously. (Going the other way round is also possible but requires YAML > JSON converting and a bit care.) But this way, switching between both should be just a 'stop one, start the other' (possibly updating / copying config first).

Hope this helps!

**Comment by james-fry** [comment]
> The disadvantage is mainly that you are depending on the release cycle of the addon (although this is normally the same as for Z2M). 

There is an edge version of the addon that installs the latest dev z2m.
Downside is that to upgrade you need to delete the addon and reinstall. (Needing to copy/paste the config ofc)

**Comment by CodeFinder2** [comment]
> > The disadvantage is mainly that you are depending on the release cycle of the addon (although this is normally the same as for Z2M).
> 
> There is an edge version of the addon that installs the latest dev z2m.
> Downside is that to upgrade you need to delete the addon and reinstall. (Needing to copy/paste the config ofc)

Yes, but (although this is solved now): https://github.com/danielwelch/hassio-zigbee2mqtt/issues/284

I wrote that because one should consider that there's the "addon in between", adding an additional/potential source of errors.

**Comment by TheStigh** [comment]
@CodeFinder2 @james-fry 

Thanks to both of you for the answers!
I already use Portainer on my RPi4 running the test-installation of z2m together with z2m-assistant.

May I ask another question;
My house do have several floors and there is no way my Coordinator will reach one particular floor as it is concrete in all walls. The only thing I have there that reaches the rest of the house are cabled Cat6 network. My plan was to set my RPi4 on this floor with a CC2531, but the question is then, what firmware should I use on this so it communicates correct with HA/z2m (with CC1352P-2) on the main floor? Or am I totally off understanding how z2m works?

PS; my CC1352P-2 arrives next week - is there any difference in setting it up on z2m addon vs z2m standalone? Anything I should be aware of (hoping I don't get a deprecated revision, but it is coming directly from TI).

**Comment by TheStigh** [comment]
> 
> 
> > [...]
> > So a nice little bit of improvement there, I should get an SMA antenna to increase that even more, anyone can recommend me one? [...]
> 
> I bought this one (because it was also labeled "Zigbee") and it works good, see my post above: https://www.digikey.de/product-detail/de/linx-technologies-inc/ANT-2.4-CW-HW-SMA/ANT-2.4-CW-HW-SMA-ND/2391322 (8€)
> 
> Don't know if any similar (Cheap Chinese 😂) SMA 2.4 GHz antenna would do the same job. 😁

If you already have paired devices to the CC1352P-2 and then do the mod/add the external antenna, do you have to re-pair afterwards? Or is that part 'plug-and-play' after re-soldering the capacitor?

**Comment by CodeFinder2** [comment]
> @CodeFinder2 @james-fry
> 
> Thanks to both of you for the answers!
> I already use Portainer on my RPi4 running the test-installation of z2m together with z2m-assistant.
> 
> May I ask another question;
> My house do have several floors and there is no way my Coordinator (or any powered zigbee devices) will reach one particular floor as it is concrete in all walls. The only thing I have there that reaches the rest of the house are cabled Cat6 network. My plan was to set my RPi4 on this floor with a CC2531, but the question is then, what firmware should I use on this so it communicates correct with HA/z2m (with CC1352P-2) on the main floor? Or am I totally off understanding how z2m works?

If you are sure that it's impossible to get any Zigbee signal to that floor, you can surely use your RPi (connected via Ethernet) and create a separate Zigbee network (in addition to the one created by your CC1352P-2 coordinator). On the RPi would then run Z2M as well, publishing messages to the same MQTT broker (via Ethernet). The CC2531 would then be a coordinator as well (but spawning a DIFFERENT network). Although I have never used this kind of setup myself but you are welcome to report your experiences here! :-)

> PS; my CC1352P-2 arrives next week - is there any difference in setting it up on z2m addon vs z2m standalone? Anything I should be aware of (hoping I don't get a deprecated revision, but it is coming directly from TI).

Not really. Except that the addon requires JSON and Z2M requires YAML configuration. Also "port: null" does not work in the Hassio addon. According to my knowledge, TI only sends the recent revisions so no need to worry.

**Comment by CodeFinder2** [comment]
> > > [...]
> > > So a nice little bit of improvement there, I should get an SMA antenna to increase that even more, anyone can recommend me one? [...]
> > 
> > 
> > I bought this one (because it was also labeled "Zigbee") and it works good, see my post above: https://www.digikey.de/product-detail/de/linx-technologies-inc/ANT-2.4-CW-HW-SMA/ANT-2.4-CW-HW-SMA-ND/2391322 (8€)
> > Don't know if any similar (Cheap Chinese 😂) SMA 2.4 GHz antenna would do the same job. 😁
> 
> If you already have paired devices to the CC1352P-2 and then do the mod/add the external antenna, do you have to re-pair afterwards? Or is that part 'plug-and-play' after re-soldering the capacitor?

I have started the pairing (re-pairing because I used a CC2531 previously) after I did the mod but if you do the pairing before the mod, no re-pairing should be required. You may want to keep in mind that IF you damage the board during the re-soldering (which is VERY unlikely), you have to go back to the CC2531 (= repairing). However, it's virtually impossible to destroy the board although it can happen that you destroy the capacitor because its very tiny and it depends on your soldering skills. Nonetheless, this isn't a real problem as well since they are really cheap. ;-)

**Comment by remkolems** [comment]
I did the the capacitor mod and was not successful. I destroyed the capacitor completely. Signal became half of CC2531 (oops), So watch out! I had to buy a new board to continue my journey....

Planning to repair my board, once I have the correct soldering station and miscellaneous required tools. I will keep this thread up to date. 

See also:

(https://zigbee2mqtt.discourse.group/t/cc1352p-switch-to-external-antenna-instead-of-pcb/681/6?u=rlems)
and
https://zigbee2mqtt.discourse.group/t/cc1352p-switch-to-external-antenna-instead-of-pcb/681/7?u=rlems

**Comment by timdonovanuk** [comment]
I really don't understand the design. Why even bother putting a SMA connector on...but then require phsyical modification to the board to actually support it. facepalm.

**Comment by james-fry** [comment]
> I really don't understand the design. Why even bother putting a SMA connector on...but then require phsyical modification to the board to actually support it. facepalm.

Because its a hardware dev kit not an "end user" device.
TI are expecting this to be used to develop new devices - hence assumption there will be support engineers well able to resolder a SMD device.

**Comment by timdonovanuk** [comment]
I understand that, but thats not really a good justification, IMO. They also could have just shipped the entire board as a bag of parts and expect Ikea style self assembly? 😆

There are various jumpers on the board that allow changes, rather than having to solder a trace on..same principle.

**Comment by james-fry** [comment]
> I understand that, but thats not really a good justification, IMO. They also could have just shipped the entire board as a bag of parts and expect Ikea style self assembly? 😆
> 
> There are various jumpers on the board that allow changes, rather than having to solder a trace on..same principle.

I thought on this too when I 1st saw the design.
I'm no RF engineers, but I think the losses and EMC from adding a jumper in the signal path would be unacceptable.

**Comment by CodeFinder2** [comment]
@remkolems Sorry to hear that and thanks for sharing your insights with us! That's why I warned about the mod/resoldering. 😑

Hope you can finally repair it!

**Comment by fredrikgk** [comment]
> I understand that, but thats not really a good justification, IMO. They also could have just shipped the entire board as a bag of parts and expect Ikea style self assembly? 😆

There are various jumpers on the board that allow changes, rather than having to solder a trace on..same principle.

You cannot use a conventional jumper for RF signals, that will introduce impedance mismatch and RF signal loss. 

Similarly, you cannot simply split the RF path in two. Even when using a proper RF design splitter, which can be implemented on PCB, you would get 3 dB loss. That is not acceptable.

The design is made for TI´s customers to easily be able to evaluate the radiated performance out of the box. Thus, the PCB antennas are connected by default. The SMA is intended primarily for customers who want to do advanced RF testing by connecting the radio directly to RF instruments (spectrum analyser, RF signal generator, etc.). These are typically fully capable of soldering a 0402 component. 

@remkolems , the simplest way to solder the capacitor is to use a hot-air soldering iron. Combine with generous amounts of liquid flux and a good microscope and it is quite easy.

**Comment by patrickharris** [comment]
Thanks guys for the clue to press the soft reset button next tot the USB connector. So happy it's working now.

**Comment by valepe** [comment]
I flashed my CC1352P-2 and I'm ready to change the panID and the channel (avoid wifi interferences) so I must repair my devices (no problem).
But I have a question: after paired the devices z2m will change the actual ID (eg 0x....) of my devices?

**Comment by CodeFinder2** [comment]
@valepe No, those ID won't change.

**Comment by valepe** [comment]
> @TASSDevon: no, it does **not auto-detect an external antenna**. You have to **resolder a tiny capacitor** on the board (CC1352P-2) to use the external antenna...unfortunately.

Can you tell me how you move this capacitor? Did you use a hot air station or a solder tip?
Thanks

**Comment by CodeFinder2** [comment]
With a fine solder tip ;-)

**Comment by valepe** [comment]
> With a fine solder tip ;-)

Did it in a breaveheart moment. :)
I used and edge tip but wtf these 0402 components are too small for my tastes...

**Comment by alekslyse** [comment]
Did the need to reset CC1352P-2/CC26X2R1 ever get solved? Is it possible to solve this in firmware or was it a really bad choice to buy those boards when it randomly require a reset?

**Comment by TASSDevon** [comment]
> Did the need to reset CC1352P-2/CC26X2R1 ever get solved? Is it possible to solve this in firmware or was it a really bad choice to buy those boards when it randomly require a reset?

Mine has been running for months without a reset required.

**Comment by CodeFinder2** [comment]
> > Did the need to reset CC1352P-2/CC26X2R1 ever get solved? Is it possible to solve this in firmware or was it a really bad choice to buy those boards when it randomly require a reset?
> 
> Mine has been running for months without a reset required.

Dito 😏

**Comment by alekslyse** [comment]
Strange. Om raspberry pi or nuc/pc. I'm running on a nuc and seem to need a reset from time to time. From what I understand it has not been a problem on pi's

**Comment by TheStigh** [comment]
> 
> 
> Strange. On raspberry pi or nuc/pc? I'm running on a nuc and seem to need a reset from time to time. From what I understand it has not been a problem on pi's

Only thing I do experience is that my NUC i7 on reboot, about every 3-5th reboot I need to power-cycle the CC, but on my NUC i5 that never happens. Either it is an USB issue, or one of my CC's are faulty. I could of course try to re-flash but it hasn't been that big issue to me :)

**Comment by TASSDevon** [comment]
Mine is hooked up to a Beaglebone which is never rebooted

**Comment by presslab-us** [comment]
> Did the need to reset CC1352P-2/CC26X2R1 ever get solved? Is it possible to solve this in firmware or was it a really bad choice to buy those boards when it randomly require a reset?

I messed around with a bunch of stuff to try and fix this.  Using an oscilloscope I have seen that when it fails, the CC1352P is trying to send data back to the PC, but the PC never receives it.  The XDS110 debugger (the on-board one) somehow gets into a state where it doesn't relay this data.

What seems to have solved it is updating the XDS110 to the latest version.  The firmware is included here:
https://software-dl.ti.com/ccs/esd/documents/xdsdebugprobes/emu_xds_software_package_download.html

And instructions; use the path "C:\ti\ccs_base\common\uscif\xds110":
http://software-dl.ti.com/ccs/esd/documents/xdsdebugprobes/emu_xds110.html#updating-the-firmware

It used to need a reset every time I reboot the PC, but now I have not needed to reset it and I've rebooted the PC at least a half dozen times.

**Comment by james-fry** [comment]
> > Did the need to reset CC1352P-2/CC26X2R1 ever get solved? Is it possible to solve this in firmware or was it a really bad choice to buy those boards when it randomly require a reset?
> 
> I messed around with a bunch of stuff to try and fix this. Using an oscilloscope I have seen that when it fails, the CC1352P is trying to send data back to the PC, but the PC never receives it. The XDS110 debugger (the on-board one) somehow gets into a state where it doesn't relay this data.
> 
> What seems to have solved it is updating the XDS110 to the latest version. The firmware is included here:
> https://software-dl.ti.com/ccs/esd/documents/xdsdebugprobes/emu_xds_software_package_download.html
> 
> And instructions; use the path "C:\ti\ccs_base\common\uscif\xds110":
> http://software-dl.ti.com/ccs/esd/documents/xdsdebugprobes/emu_xds110.html#updating-the-firmware
> 
> It used to need a reset every time I reboot the PC, but now I have not needed to reset it and I've rebooted the PC at least a half dozen times.

Nice analysis :) 
Thanks

**Comment by valepe** [comment]
> > Did the need to reset CC1352P-2/CC26X2R1 ever get solved? Is it possible to solve this in firmware or was it a really bad choice to buy those boards when it randomly require a reset?
> 
> I messed around with a bunch of stuff to try and fix this. Using an oscilloscope I have seen that when it fails, the CC1352P is trying to send data back to the PC, but the PC never receives it. The XDS110 debugger (the on-board one) somehow gets into a state where it doesn't relay this data.
> 
> What seems to have solved it is updating the XDS110 to the latest version. The firmware is included here:
> https://software-dl.ti.com/ccs/esd/documents/xdsdebugprobes/emu_xds_software_package_download.html
> 
> And instructions; use the path "C:\ti\ccs_base\common\uscif\xds110":
> http://software-dl.ti.com/ccs/esd/documents/xdsdebugprobes/emu_xds110.html#updating-the-firmware
> 
> It used to need a reset every time I reboot the PC, but now I have not needed to reset it and I've rebooted the PC at least a half dozen times.

Great!
Can I update the XDS110 with Uniflash?
Or should I use the support utilities?

**Comment by CodeFinder2** [comment]
@presslab-us from me as well, great analysis and thanks for sharing! 🙂

**Comment by presslab-us** [comment]
> Great!
> Can I update the XDS110 with Uniflash?
> Or should I use the support utilities?

@valepe I'm not sure if Uniflash can program the XDS110, or what version it comes with.  You can try it and let us know.  The version I programmed from my link above was `firmware_3.0.0.11.bin`

**Comment by alekslyse** [comment]
I got it updated to .11 by uniflash (it said it had a new fw when I was about to upload the new z2m firmware), but got 03.00.00.13 from the link above

**Comment by alekslyse** [comment]
Also remember the port of change on the update so z2m won't change if you don't update the path. The version is in the name for example     /dev/serial/by-id/usb-Texas_Instruments_XDS110__03.00.00.13__Embed_with_CMSIS-DAP_L430021O-if00

**Comment by valepe** [comment]
> I got it updated to .11 by uniflash (it said it had a new fw when I was about to upload the new z2m firmware), but got 03.00.00.13 from the link above

I updated with Uniflash but I can't find where to read the XDS110 firmware version.
I found in System report -> USB -> XDS110 (3.0.0.12) so I suppose that this is the firmware version just updated.

**Comment by upwindanderl** [comment]
**I did a fw update tonight and still have this reboot / reset problem :(** 

Either sometimes it's pretty hard to get the board running. It's not only pressing the reset button once and everything is ok. Supershitty :(

**Comment by presslab-us** [comment]
In my case I could always unplug and replug the USB and it would be fine.  Does that work for you? 
 I didn't use the reset button because it didn't work well in the case I made.
Since updating the XDS110 I have rebooted many times and there has been no problem.
Maybe you can try removing the jumpers for TCK and TMS?

**Comment by upwindanderl** [comment]
> In my case I could always unplug and replug the USB and it would be fine. Does that work for you?
> I didn't use the reset button because it didn't work well in the case I made.
> Since updating the XDS110 I have rebooted many times and there has been no problem.
> Maybe you can try removing the jumpers for TCK and TMS?

hmmmm sounds promising! 👍 

The thing is the server is not nearby me and i'm only remote on it. As well plugging or unplugging is a total difficult phonecall and some remote guidance. 
The only thing i know is that i updated the last FW yesterday which was automatically recommended with uniflash  as i've been there and again today big trouble after rebooting the server. 
Next time im there i'll check exactly the fw version and what you mean with tck and tms :-)

**Comment by upwindanderl** [comment]
> 
> 
> > In my case I could always unplug and replug the USB and it would be fine. Does that work for you?
> > I didn't use the reset button because it didn't work well in the case I made.
> > Since updating the XDS110 I have rebooted many times and there has been no problem.
> > Maybe you can try removing the jumpers for TCK and TMS?
> 
> hmmmm sounds promising! 👍
> 
> The thing is the server is not nearby me and i'm only remote on it. As well plugging or unplugging is a total difficult phonecall and some remote guidance.
> The only thing i know is that i updated the last FW yesterday which was automatically recommended with uniflash as i've been there and again today big trouble after rebooting the server.
> Next time im there i'll check exactly the fw version and what you mean with tck and tms :-)

So in the end it looks pretty solid! 
Had lot reboots because of server problems and of course some more restarts of HA and i can't remember to push the button again.
Didn't changed anything on the jumpers, the only solution for me to click yes after i was asked for some firmware update of the UNIFLASH tool

**Comment by likellindil** [comment]
> > > In my case I could always unplug and replug the USB and it would be fine. Does that work for you?
> > > I didn't use the reset button because it didn't work well in the case I made.
> > > Since updating the XDS110 I have rebooted many times and there has been no problem.
> > > Maybe you can try removing the jumpers for TCK and TMS?
> > 
> > 
> > hmmmm sounds promising! 👍
> > The thing is the server is not nearby me and i'm only remote on it. As well plugging or unplugging is a total difficult phonecall and some remote guidance.
> > The only thing i know is that i updated the last FW yesterday which was automatically recommended with uniflash as i've been there and again today big trouble after rebooting the server.
> > Next time im there i'll check exactly the fw version and what you mean with tck and tms :-)
> 
> So in the end it looks pretty solid!
> Had lot reboots because of server problems and of course some more restarts of HA and i can't remember to push the button again.
> Didn't changed anything on the jumpers, the only solution for me to click yes after i was asked for some firmware update of the UNIFLASH tool

Hi there,

My HA & Z2M are built on my NUC and I met exactly the same issue.
May I know how you fixed this issue in the end?
Now I have to unplug and re-plug it every one or two days to keep it working.

**Comment by joshuakoh1** [comment]
Has this been fixed yet?

**Comment by timdonovanuk** [comment]
My personal experience, coming from a CC2531 to the CC1352p-2 is the CC1352p-2 has been a *nightmare*. Weekly I get devices drop off my network, that need repairing. If I stop and restart Z2M I quite often need to repress the button on the side of the CC1352P, which is not great as physical access to it is not easy (it's up in my loft). 

I tried to flash the CC1352P today using the command line TI tools and it actually managed to take out my entire Proxmox instance!

**Comment by joshuakoh1** [comment]
@timdonovanuk Yup I'm getting really upset about th need to press the reset button, I just wish this issue was made more obvious in the supported sticks page. I would not have purchased this at all if I knew about it from the start. Open to suggestions about a better stick without this issue as well.

Separately, how did you manage to pass through the device to VM on proxmox? Passing the USB port doesn't create any /dev/ttyACM0 on the VM.

**Comment by timdonovanuk** [comment]
I have to agree, based on my personal experience I don't think CC1352P should be in the list of recommended devices, but maybe it's just some of us. Today I'm experiencing more fun, some of my bulbs randomly dont work with `Data request failed with error: 'No network route' (205)` errors 🤦 

@joshuakoh1 here are my proxmox settings, works fine for me:

![image](https://user-images.githubusercontent.com/8156439/102205943-c9cafb80-3ec3-11eb-9aae-51aa166b3087.png)

Needs a hard reboot. Then I get these showing up:

```
~$ ls -l /dev/serial/by-id
total 0
lrwxrwxrwx 1 root root 13 Dec 14 15:57 usb-Texas_Instruments_XDS110__03.00.00.05__Embed_with_CMSIS-DAP_L43001PG-if00 -> ../../ttyACM0
lrwxrwxrwx 1 root root 13 Dec 14 15:57 usb-Texas_Instruments_XDS110__03.00.00.05__Embed_with_CMSIS-DAP_L43001PG-if03 -> ../../ttyACM1

```

**Comment by joshuakoh1** [comment]
@timdonovanuk What distro are you using btw? Mine was Ubuntu 20.04 and there's no /dev/serial at all

**Comment by timdonovanuk** [comment]
@joshuakoh1 Debian 10. To be fair, I think if the adapter is not passed through successfully then I used to get no /dev/serial at all either. Glad you got it going!

**Comment by timdonovanuk** [comment]
I finally updated to the latest firmware on the adapter and it's been an unstable mess. Once a day a different bulb drops off the network, or a battery powered sensor or button has to be repaired. It's actually worse than the firmware I was using before (~6 months old). At this point I don't know whether to abandon z2m or abandon the CC1352.

**Comment by joshuakoh1** [comment]
Looking for alternatives to CC1352P-2. Really frustrated with the stupid reset button problem. Does zzh/slaesh's stick have this issue?

**Comment by timdonovanuk** [comment]
@joshuakoh1 FWIW throughout all of this, my logs have been showing a lot of "failed to ping <device>". Today I updated to the latest version 1.17 and also set `availability_timeout: 0` (it used to be set to 60) and so far not a single error in my log. I found the advice about `availability_timeout `in another issue thread. For some people setting it to zero made z2m more stable, and for others setting it to 60 made it more stable so I don't know...but so far so good.

**Comment by silver0007** [comment]
I see that issue is closed but is it fixed anyhow?
I moved over a week ago from PRi to NUC8i3BEH which runs HassOS in VM and I struggle with same error as everyone above. My CC1352 just disconnects for no reason. Once a day, sometimes even 3 times a day... Fast option to bring back connection is to re-plug USB.
I updated firmware to the newest available for CC1532, but didn't help. Had no issues with it on RPi (worked perfect for almost a year), so clearly looks like NUC problem as discussed above.
Yesterday I changed availability_timeout to "60" from default which is "0" I believe but didn't help too. Got same error after 11 hrs of running. And in my case problem do not occur only after system reboot etc., it just keeps getting disconnected during regular runtime and re-plugging USB twice a day is really annoying... Did anyone found permanent solution?

**Comment by presslab-us** [comment]
I had problems before where I would need to replug it after rebooting the VM.  New firmware seemed to fix that problem.

Then I had problems where it would disconnect after about a week of running.  That problem seems to be fixed now, my CC1352P-2 has been stable for a few months now, no problems at all.  I believe the improvement was related to upgrading my qemu (now running QEMU emulator version 5.2.0 (Debian 1:5.2+dfsg-3)).  I'm running on Debian Sid.

**Comment by noohi** [comment]
Mine is also now much better, potentially fixed.

What I think helped the most is to update the debugger, basically run SmartRF Flash Programmer 2 and it should popup the recommended debugger upgrade - it's parallel to the actual app/firmware from Z2M.

**Comment by silver0007** [comment]
@noohi, so what you say is to install "SmartRF Flash Programmer 2" and then after plugging CC1352 it will recognize that upgrade is recommended and run it automatically if I confirm?
I'm asking as that sounds quite safe (safer at least than downloading firmware by myself somewhere from the web) and I want to make sure that I won't do anything that breaks my system at this point... Have maybe around 60 zigbee devices so in case of a need of re-pairing etc. it's just a nightmare.

**Comment by noohi** [comment]
@silver0007 , yes, it will look like [this](https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQtwJDogoHPyFf8Qpv7zRgKOf0R2kCuqJMKXQ&usqp=CAU), page 6 in the [manual](https://www.ti.com/lit/ug/snou148a/snou148a.pdf?ts=1613062075196&ref_url=https%253A%252F%252Fwww.ti.com%252F).

**Comment by silver0007** [comment]
Thanks! I’ll try if it helps.

**Comment by silver0007** [comment]
Downloaded Flash Programmer 2 but unfortunately no pop up's about automatic upgrade.
![image](https://user-images.githubusercontent.com/61536693/108058532-78babd80-7054-11eb-8ee9-409ef728a4ae.png)

Looks that CC1352 is recognized correctly. There's "Unknown" on the list probably for debugger. Not sure if there's anything that I should upload manually. Anyone did that? I'm afraid to experiment as I have so many hidden zigbee devices that I'm terrified that I'll need to re-pair them if something goes wrong...
On the other hand, re-plugging USB twice a day is also not the most convenient process ever...

**Comment by silver0007** [comment]
@Koenkk, this issue is closed so there won't be any further investigations about the root cause and potential fix?

I'm thinking about changing adapter to CC2652R, but the question is: will I have to re-pair devices?

**Comment by Koenkk** [comment]
@silver0007 I doubt if the issue can be solved from z2m, likely it's a combination of adapter/OS/computer. I don't know how we can debug this further.

> I'm thinking about changing adapter to CC2652R, but the question is: will I have to re-pair devices?

No, https://www.zigbee2mqtt.io/information/FAQ.html#requires-repairing

> Except when switching between adapters with the following chips: CC2652, CC1352, CC253* (only when running Zigbee 3.0 firmware)

**Comment by silver0007** [comment]
Thanks for info! I'll try with Slaesh's CC2652RB and see if that helps.
Additionally I think it would be good to add some warning about CC1352 on recommended devices page:
https://www.zigbee2mqtt.io/information/supported_adapters.html
Now it's only mentioned in FAQ which everyone check when it's already too late.

**Comment by silver0007** [comment]
Just to summarize my story - I bough CC2652P stick to replace CC1352P launchpad. New adapter works without any issues for over 4 days and as said above, I didn't have to repair any devices.
Range on new stick is quite similar as on CC1352, means equally good as I've never had any range issues with CC1352 in 180m2 house.

**Comment by silver0007** [comment]
Just to summarize my story - I bough CC2652P stick to replace CC1352P launchpad. New adapter works without any issues for over 4 days and as said above, I didn't have to repair any devices.
Range on new stick is quite similar as on CC1352, means equally good as I've never had any range issues with CC1352 in 180m2 house.

**Comment by valepe** [comment]
> Just to summarize my story - I bough CC2652P stick to replace CC1352P launchpad. New adapter works without any issues for over 4 days and as said above, I didn't have to repair any devices.
> Range on new stick is quite similar as on CC1352, means equally good as I've never had any range issues with CC1352 in 180m2 house.

Interesting. But which power level did you use with CC1352?

**Comment by noohi** [comment]
Aa a side-note, which may be relevant to those with intermittent disconnects of USB on AMD systems:
[AMD Finds a Fix for Ryzen's USB Problems, BIOS Patch Coming in Early April](https://www.tomshardware.com/amp/news/amd-issuing-usb-fix-for-ryzen-chips-bios-patch-coming-in-early-april)

**Comment by timdonovanuk** [comment]
My CC1352P-2 actually died the other night. I replaced it with an old CC2531 and with the normal firmware - it kept crashing. I then flashed it with the "source routing" (i.e. stable firmware used with big networks) and introduced a second CC2531 repeater. You can roll back from CC1352P-2 -> CC2531 without repairing devices (just had to delete coordinator_backup.json).

Since I've done this, my network has never been more stable. Maybe it was the introduction of the CC2531 repeater, but given the antenna size on the CC1352P-2 it shouldn't have been necessary. I will get a ZZH when I can, but I'm in no hurry to ever use the CC1352P-2 again (I appreciate technically the ZZH uses the same chip but lets see).

**Comment by jvrrr** [comment]
I run this CC1352P-2 for more then a year and never had issue. I run zigbee2mqtt in container on Ubuntu on VMWare ESXI on a HPe Microserver 10. I still need to find a case for my CC1352...Maybe the case is giving issues in your situation)

The only time I had a issue was this:

I use  Osram "smart+ plug" for a couple of lamps.
One day I decided not to use one plug anymore.
But there where some script still running in Domoticz that wanted to switch the Osram.
and I also was forget to remove te device in Zigbee2mqtt.

This situation did run well for about 20 hours and then finally the CC1352P crashes and I needed to reset it,
Day in day out, I needed to restart zigbee2mqtt and reset the CC1352P
After weeks I realisted I removed the Osram hard from the network, so I booted the Osram again and after that the CC1352p runs well for months now.

Maybe my story can help you out.

I'm realy happy with this CC1352P-2!
